### ISSUE TITLE
Question: What does it mean to use optimizer with addnoise=False and where is KSGLD implementation?

### ISSUE THREAD

**Issue by har07** [henripal/sgld#3]
First of all, thanks for the paper and the source code! Both had influenced the direction of the research topic for my master degree (still in early phase).

KSGLD is mentioned in the paper but I couldn't find the code for KSGLD optimizer anywhere. Did it implemented in some other repository or did I miss something?

Another question (or more of a request for clarification), what does it mean to use SGLD/pSGLD with parameter `addnoise=False`? I guess:
-  `SGLD` optimizer with `addnoise=False` is equivalent to running classic SGD optimizer. I tried to verify this by using this `SGLD` without noise, and compare the result to using PyTorch's SGD optimizer. And the accuracy in each epoch for both optimizer came out almost identical. So I'm almost sure this interpretation is correct, just want to confirm.
-  `pSGLD` optimizer with `addnoise=False` is equivalent to running classic SGD with RMSProp, or just RMSProp for short, is this correct? ~(I haven't tried to compare `pSGLD` without noise to PyTorch's RMSProp optimizer)~. I have just tried training the model in this repo using PyTorch's RMSprop and the accuracy in each epoch also identical to using pSGLD without noise.

**Comment by henripal** [comment]
Hi - sorry for the long response time. Haven't checked this repo out in a while. KSGLD I think I used the `tf.contrib` implementation of KFAC. It isn't in this repo, sorry.

For your other questions, yes, your understanding is correct.

Thanks for your interest and let me know if you have any other questions!

**Comment by har07** [comment]
Thanks for your reply! Probably you implemented KSGLD in this other repo: [sgld_tf](https://github.com/henripal/sgld_tf/tree/master/sgld_tf)? I didn't realize it's there, will try to study the codes now

**Comment by har07** [comment]
Not sure if this is the right place to discuss this, but I have another question. I'm using this repository as a baseline code trying to reproduce the result reported in your paper, but I couldn't get close. pSGLD and SGLD need more epochs (more than 100) to achieve decent accuracy compared to SGD and RMSProp. Do you keep notes on what was the hyperparameters, learning rate and others? 

I used the following learning rates to get the result plotted at the bottom:
- SGD: 0.1​
- RMSProp: 0.0005​
- SGLD: 0.0005​
- pSGLD: 0.99​ (other hyperparams were defult: `alpha=0.99, eps=1e-8, centered=False, addnoise=True`)

![image](https://user-images.githubusercontent.com/11190794/105703093-521ad280-5f3f-11eb-98fd-a108e089307d.png)

![image](https://user-images.githubusercontent.com/11190794/105703123-59da7700-5f3f-11eb-8aeb-71092dea424c.png)

Thanks for your time!
